core/src/main/java/org/jruby/truffle/translator/BodyTranslator.java

```diff
@@ -346,9 +351,124 @@ public RubyNode visitBreakNode(org.jruby.ast.BreakNode node) {
+            arguments.add(new SelfNode(context, sourceSection));
+            argumentsCount--;
+        }
+
+        for (int n = 0; n < argumentsCount; n++) {
+            arguments.add(new ReadPreArgumentNode(context, sourceSection, n, MissingArgumentBehaviour.UNDEFINED));
+        }
+
+        return new CallRubiniusPrimitiveNode(context, sourceSection,
+                primitive.getFactory().createNode(context, sourceSection, arguments.toArray(new RubyNode[arguments.size()])),
+                environment.getReturnID());
+    }
+
+    private RubyNode translateRubiniusInvokePrimitive(SourceSection sourceSection, CallNode node) {
+        /*
+         * Translates something that looks like
+         *
+         * Rubinius.invoke_primitive :foo, arg1, arg2, argN
+         *
+         * into
+         *
+         * CallRubiniusPrimitiveNode(FooNode(arg1, arg2, ..., argN))
+         */
+
+        if (node.getArgsNode().childNodes().size() < 1 || !(node.getArgsNode().childNodes().get(0) instanceof org.jruby.ast.SymbolNode)) {
+            throw new UnsupportedOperationException("Rubinius.invoke_primitive must have at least an initial literal symbol argument");
+        }
+
+        final String primitiveName = ((org.jruby.ast.SymbolNode) node.getArgsNode().childNodes().get(0)).getName();
+
+        final RubiniusPrimitiveConstructor primitive = context.getRubiniusPrimitiveManager().getPrimitive(primitiveName);
+
+        final List<RubyNode> arguments = new ArrayList<>();
+
+        for (org.jruby.ast.Node arg : node.getArgsNode().childNodes().subList(1, node.getArgsNode().childNodes().size())) {
+            arguments.add(arg.accept(this));
+        }
+
+        return new CallRubiniusPrimitiveNode(context, sourceSection,
+                primitive.getFactory().createNode(context, sourceSection, arguments.toArray(new RubyNode[arguments.size()])),
+                environment.getReturnID());
+    }
+
+    private RubyNode translateRubiniusPrivately(SourceSection sourceSection, CallNode node) {
+        /*
+         * Translates something that looks like
+         *
+         * Rubinius.privately { foo }
+         *
+         * into
+         *
+         * foo
+         *
+         * While we translate foo we'll make all call sites as ignoring visbility.
+         */
+
+        if (!(node.getIterNode() instanceof org.jruby.ast.IterNode)) {
+            throw new UnsupportedOperationException("Rubinius.privately needs a literal block");
+        }
+
+        if (node.getArgsNode() != null && node.getArgsNode().childNodes().size() > 0) {
+            throw new UnsupportedOperationException("Rubinius.privately should not have any arguments");
+        }
+
+        currentCallMethodName = "privately";
+        privately = true;
+
+        try {
+            return (((org.jruby.ast.IterNode) node.getIterNode()).getBodyNode()).accept(this);
+        } finally {
+            privately = false;
+        }
```

Comment:
This scoping with the privately variable could be described in a comment. What are you trying to achieve here?